src/main/java/com/github/wovnio/wovnjava/Headers.java

```diff
@@ -211,6 +211,62 @@ String redirectLocation(String lang) {
         }
     }
 
+    public String locationWithLangCode(String location) {
+        // check if needed
+        if (location == null) {
+            return null;
+        }
+        if (langCode().equals(settings.defaultLang)) {
+            return location;
+        }
+
+        // catprue protocl and path
+        String locationProtocol = protocol;
+        String path;
+        if (location.contains("://")) {
+            if (!location.contains("://" + host)) {
+                return location;
+            }
+            String[] protocolAndRemaining = location.split("://", 2);
+            locationProtocol = protocolAndRemaining[0];
+            path = "/" + protocolAndRemaining[1].split("/", 2)[1];
+        } else {
+            path = location.startsWith("/") ? location : pathJoin("/", pathJoin(pathNameKeepTrailingSlash, location));
+        }
+        path = pathNormalize(path);
+
+        // check location already have language code
+        if (settings.urlPattern.equals("query") && location.contains("wovn=")) {
+            return location;
+        } else if (settings.urlPattern.equals("path")) {
+            Pattern p = Pattern.compile(settings.urlPatternReg);
+            Matcher m = p.matcher(path);
+            if (m.find()) {
+                String l = m.group(1);
+                if (l != null && l.length() > 0 && Lang.getLang(l) != null) {
+                    return location;
+                }
+            }
+        }
+
+        // build new location
+        String lang = langCode();
+        String queryLangCode = "";
+        String subdomainLangCode = "";
+        String pathLangCode = "";
+        if (settings.urlPattern.equals("query")) {
+            if (location.contains("?")) {
+                queryLangCode = "&wovn=" + lang;
+            } else {
+                queryLangCode = "?wovn=" + lang;
+            }
+        } else if (settings.urlPattern.equals("subdomain")) {
+            subdomainLangCode = lang + ".";
+        } else {
+            pathLangCode = "/" + lang;
+        }
+        return locationProtocol + "://" + subdomainLangCode + host + pathLangCode + path + queryLangCode;
+    }
```

Comment:
What is the difference between
```
String redirectLocation(String lang) {...}
```
and
```
public String locationWithLangCode(String location) {...}
```
To me, they seem to try to do the same thing.

Reply:
> redirectLocation
This method is to create new URL from browser URL and specific language code.
borwser URL: https://foo.com/
language code: fr
result: https://foo.com/fr/

> locationWithLangCode
This method is to create new URL from specific URL and current language code.
borwser URL: https://foo.com/fr/signin
specific URL: /home
result: https://foo.com/fr/home

Reply:
Gotcha. Thanks for the examples.